src/app/archive_hardfork_toolbox/archive_hardfork_toolbox.ml

```diff
@@ -0,0 +1,138 @@
+(* berkeley_migration_verifier.ml -- verify integrity of migrated archive db from original Mina mainnet schema *)
+
+open Async
+open Logic
+
+let run_check_and_exit check_fn =
+  let open Deferred.Let_syntax in
+  let%bind results = check_fn () in
+  report_all_checks results ;
+  if has_failures results then Shutdown.exit 1 else Deferred.return ()
+
+let is_in_best_chain_command =
+  Async.Command.async ~summary:"Verify fork block is in best chain"
+    (let open Command.Let_syntax in
+    let%map archive_uri =
+      Command.Param.flag "--archive-uri"
+        ~doc:"URI URI for connecting to the mainnet archive database"
+        Command.Param.(required string)
+    and fork_state_hash =
+      Command.Param.flag "--fork-state-hash" ~aliases:[ "-fork-state-hash" ]
+        Command.Param.(required string)
+        ~doc:"String Hash of the fork state"
+    and fork_height =
+      Command.Param.flag "--fork-height" ~aliases:[ "-fork-height" ]
+        Command.Param.(required int)
+        ~doc:"Int Height of the fork block"
```

Comment:
I assume this is the first block of fork chain? Could we clarify it here in doc?